src/main/java/uk/gov/hmcts/reform/managecase/service/NoticeOfChangeService.java

```diff
@@ -0,0 +1,229 @@
+    private static final String CASE_ROLE_ID = "CaseRoleId";
+
+    private final DataStoreRepository dataStoreRepository;
+    private final DefinitionStoreRepository definitionStoreRepository;
+    private final SecurityUtils securityUtils;
+
+    @Autowired
+    public NoticeOfChangeService(DataStoreRepository dataStoreRepository,
+                                 DefinitionStoreRepository definitionStoreRepository,
+                                 SecurityUtils securityUtils) {
+
+        this.dataStoreRepository = dataStoreRepository;
+        this.definitionStoreRepository = definitionStoreRepository;
+        this.securityUtils = securityUtils;
+    }
+
+    public ChallengeQuestionsResult getChallengeQuestions(String caseId) {
+        ChallengeQuestionsResult challengeQuestionsResult = challengeQuestions(caseId).getChallengeQuestionsResult();
+        //Step 12 Remove the answer section from the JSON returned byGetTabContents and return success with the
+        // remaining JSOn
+
+        List<ChallengeQuestion> challengeQuestionsResponse = challengeQuestionsResult.getQuestions().stream()
+            .map(challengeQuestion -> ChallengeQuestion.builder()
+                .questionText(challengeQuestion.getQuestionText())
+                .caseTypeId(challengeQuestion.getCaseTypeId())
+                .order(challengeQuestion.getOrder())
+                .answerFieldType(FieldType.builder()
+                                     .collectionFieldType(challengeQuestion.getAnswerFieldType()
+                                                              .getCollectionFieldType())
+                                     .complexFields(challengeQuestion.getAnswerFieldType()
+                                                        .getComplexFields())
+                                     .fixedListItems(challengeQuestion.getAnswerFieldType()
+                                                         .getFixedListItems())
+                                     .regularExpression(challengeQuestion.getAnswerFieldType()
+                                                            .getRegularExpression())
+                                     .max(challengeQuestion.getAnswerFieldType().getMax())
+                                     .min(challengeQuestion.getAnswerFieldType().getMin())
+                                     .id(challengeQuestion.getAnswerFieldType().getId())
+                                     .type(challengeQuestion.getAnswerFieldType().getType())
+                                     .build())
+                .displayContextParameter(challengeQuestion.getDisplayContextParameter())
+                .challengeQuestionId(challengeQuestion.getChallengeQuestionId())
+                .build())
+            .collect(Collectors.toList());
+
+
+        return ChallengeQuestionsResult.builder().questions(challengeQuestionsResponse).build();
+    }
+
+    public NoCRequestDetails challengeQuestions(String caseId) {
```

Comment:
Do these comments need to be sanitised/updated in line with implementation ?

Reply:
done